spec/javascripts/0b_jquery_extension_spec.js

```diff
@@ -1,99 +1,247 @@
+      url = $.paramUpdate(url, param);
+      //then
+      expect(url).toEqual("i.have&test_param=test_value&non_existant_param=[object Object]")
     });
-    it("Should return false if wrong param (a Date) is given", function () {
-      expect($('#tested_div').hasClasses(new Date(123456789))).toEqual(false)
+
+    it("Should change parameter value even if url contains only parameter", function () {
+      //given
+      var url = "&alone_param=alone_value"
+      //when
+      var param = { key: "alone_param", value: "new_value" }
+      url = $.paramUpdate(url, param);
+      //then
+      expect(url).toEqual("&alone_param=new_value")
     });
-    it("Should return false if undefined is given", function () {
-      expect($('#tested_div').hasClasses(undefined)).toEqual(false)
+
+    it("Should add parameter even if url is blank", function () {
+      //given
+      var url = ""
+      //when
+      var param = { key: "new_param", value: "param_value" }
+      url = $.paramUpdate(url, param);
+      //then
+      expect(url).toEqual("&new_param=param_value")
     });
-    it("Should return false if empty String is given", function () {
-      expect($('#tested_div').hasClasses(" ")).toEqual(false)
+
+    it("Should add parameter as even if url is blank and parameter is a blank string", function () {
+      //given
+      var url = ""
+      //when
+      var param = ""
+      url = $.paramUpdate(url, param);
+      //then
+      expect(url).toEqual("&undefined=undefined")
     });
-  });
 
-  describe('$.hasAttribute', function () {
-    beforeEach(function () {
-      $(document.body).append('\
+  }
+  )
+
+  describe("$.urlParam", function () {
+    it("Should return null when url contains no parameter", function () {
+      //given
+      window.location.href = "http://clara.com";
+      //when
+      var name = ""
+      var res = $.urlParam(name)
+      //then
+      expect(res).toEqual(null)
+    });
+    it("Should return blank string when url parameter has no value", function () {
+      //given
+      window.location.href = "http://clara.com&empty_param";
+      //when
+      var name = ""
+      var res = $.urlParam(name)
+      //then
+      expect(res).toEqual("")
+    });
+    it("Should return parameter value when url parameter has a value", function () {
+      //given
+      window.location.href = "http://clara.com&empty_param=filled_value";
+      //when
+      var name = ""
+      var res = $.urlParam(name)
+      //then
+      expect(res).toEqual("filled_value")
+    });
+    it("Should return ---- when url parameter has a value", function () {
+      //given
+      window.location.href = "http://clara.com&named_parameter=param_value";
+      //when
+      var name = "named_parameter"
+      var res = $.urlParam(name)
+      //then
+      expect(res).toEqual("param_value")
+    });
+    it("Should not return url parameter value when name given is different", function () {
```

Comment:
Should not ? le "not" est de trop. C'est le même cas de test que ci-dessus, donc tu peux mettre le même texte, avec (example 1) à la fin ou (example 2)